crates/flipperzero/src/furi/message_queue.rs

```diff
@@ -0,0 +1,67 @@
+use core::ffi::c_void;
+use flipperzero_sys::furi::message_queue;
+use crate::furi::Result;
+
+/// MessageQueue provides a safe wrapper around the furi message queue primitive.
+pub struct MessageQueue<M: Sized> {
+    hnd: *const message_queue::FuriMessageQueue,
+    _marker: core::marker::PhantomData<M>,
+}
+
+impl<M: Sized> MessageQueue<M> {
+    /// Constructs a message queue with the given capacity.
+    pub fn new(capacity: u32) -> Self {
```

Comment:
We should use `usize` for "sizes", since that's the convention in the standard library so avoids a bit of casting. I was thinking of actually doing that on the C definitions, since on ARM32 `usize` = `u32`.

Reply:
Done.